### PR TITLE
net: fix build with only process

### PR DESCRIPTION
**Pull request by carllerche**
This is causing doc generation to fail.

### CODE REVIEW

File: tokio-net/src/tracing.rs [hunk 1]
```diff
@@ -25,7 +25,7 @@ macro_rules! trace {
 
 #[cfg(not(feature = "tracing"))]
 macro_rules! trace {
-    ($($arg:tt)+) => {};
+    ($($arg:tt)+) => {{}};
```
Comment (jonhoo) [inline tokio-net/src/tracing.rs]: Why are these necessary?

Comment (carllerche) [reply tokio-net/src/tracing.rs]: Otherwise, you get compile errors where `error!()` is used in an expression position. Using `()` would probably have worked as well... but `{}` seemed most appropriate.

Comment (jonhoo) [reply tokio-net/src/tracing.rs]: Oh, sure, I'm aware of the general case, just curious why you included this as a part of this change.

Comment (carllerche) [reply tokio-net/src/tracing.rs]: I discovered the issue trying to figure out why doc gen didn't pass. It came up when I added a test for compiling `tokio-net` with only the `process` feature flag.
